packages/notifi-dapp-example/src/components/NotifiSignUpButton.tsx

```diff
@@ -0,0 +1,151 @@
+
+type TargetGroupData = {
+  name: string;
+  emailAddress?: string;
+  phoneNumber?: string;
+  telegramId?: string;
+  discordId?: string;
+};
+
+//todo implement in card context
+const inputs = {};
+
+export const NotifiSignUpButton: React.FC<NotifiSignUpButtonProps> = ({
+  buttonText,
+  data,
+}) => {
+  const eventTypes = data.eventTypes;
+
+  const frontendClientLogin = useFrontendClientLogin();
+
+  const router = useRouter();
+
+  const {
+    frontendClientStatus: { isInitialized, isAuthenticated },
+    frontendClient,
+  } = useNotifiClientContext();
+
+  const { loading, setCardView, useDiscord, render, setLoading, syncFtuStage } =
+    useNotifiSubscriptionContext();
+
+  const { setIsGlobalLoading } = useGlobalStateContext();
+
+  const { formErrorMessages, formState } = useNotifiForm();
+
+  const { phoneNumber, telegram: telegramId, email } = formState;
+
+  const { email: emailErrorMessage, phoneNumber: smsErrorMessage } =
+    formErrorMessages;
+
+  const targetGroup: TargetGroupData = useMemo(
+    () => ({
+      name: 'Default',
+      emailAddress: email === '' ? undefined : email,
+      phoneNumber: isValidPhoneNumber(phoneNumber) ? phoneNumber : undefined,
+      telegramId:
+        telegramId === ''
+          ? undefined
+          : formatTelegramForSubscription(telegramId),
+      discordId: useDiscord ? 'Default' : undefined,
+    }),
+    [email, phoneNumber, telegramId, useDiscord],
+  );
+
+  const renewTargetGroups = useCallback(
+    async (targetGroup: TargetGroupData) => {
+      return frontendClient.ensureTargetGroup(targetGroup);
+    },
+    [frontendClient],
+  );
+
+  const subscribeAlerts = useCallback(
+    async (eventTypes: EventTypeConfig, inputs: Record<string, unknown>) => {
+      await renewTargetGroups(targetGroup);
+
+      return subscribeAlertsByFrontendClient(
+        frontendClient,
+        eventTypes,
+        inputs,
+      );
+    },
+    [
+      renewTargetGroups,
+      subscribeAlertsByFrontendClient,
+      frontendClient,
+      targetGroup,
+    ],
+  );
+
+  const onClick = useCallback(async () => {
+    let isFirstTimeUser = false;
+    if (isAuthenticated) {
+      await frontendClientLogin();
+      const data = await frontendClient.fetchData();
+      isFirstTimeUser = (data.targetGroup?.length ?? 0) === 0;
+    }
+    setIsGlobalLoading(true);
+    setLoading(true);
+    try {
+      let success = false;
+      if (isFirstTimeUser) {
+        const subEvents = eventTypes.filter((event) => {
+          return event.optOutAtSignup ? false : true;
+        });
+        const result = await subscribeAlerts(subEvents, inputs);
+        success = !!result;
+      } else {
+        const result = await renewTargetGroups(targetGroup);
+        success = !!result;
+      }
+
+      if (success) {
+        const newData = await frontendClient.fetchData();
+        render(newData);
+        await syncFtuStage(data.isContactInfoRequired);
+        router.push('/notifi/ftu');
```

Comment:
router.push method in nextjs 14 is async, it would case racing issue possibly. You might want to make use of the `handleRoute` method in `useRouterAsync.ts`